CONTRIBUTING.md

```diff
@@ -0,0 +1,282 @@
+and install [pip](https://pip.pypa.io/en/stable/installing/) for Python3,
+after which you can install the dependencies using:
+
+```
+yum install python-ecdsa python34-ecdsa pylint \
+    python-coverage python34-coverage python2-hypothesis \
+    python34-libs python-unittest2 python-mock python-pip epydoc
+pip2 install diff-cover
+pip3 install hypothesis diff-cover pylint
+```
+
+Optional module dependencies:
+
+* tackpy
+* m2crypto
+* pycrypto
+* gmpy
+
+On Fedora they can be installed using:
+
+```
+pip install tackpy
+dnf install m2crypto python-crypto python3-crypto python-gmpy2 python3-gmpy2
+```
+
+#### Virtual environment
+
+While the above descriptions ask for installing the packages system-wide, it's
+possible to install python packages without root permissions, that does
+require use of package like
+[`virtualenv`](https://virtualenv.pypa.io/en/stable/) or the
+[`venv`](https://docs.python.org/3/tutorial/venv.html) from Python 3.
+
+### Make changes
+
+* Fork the tlsfuzzer project
+* Clone to your local machine your fork: `git clone git@github.com:.../tlsfuzer.git`
+* Download tlslite-ng:
+  * `git clone https://github.com/tomato42/tlslite-ng.git .tlslite-ng`
+  * `ln -s .tlslite-ng/tlslite tlslite`
+* Verify that test cases are runnable: `make test`
+* In your cloned repository, create a topic branch for your upcoming patch
+  (e.g. 'implement-aria' or 'bugfix-osx-crash')
+  * usually this is based on the master branch
+  * to create branch based on master:
+    `git checkout master; git branch <example-name>` then
+    checkout the branch `git checkout <example-name>`. For your own convinience
+    avoid working directly on the `master` branch.
+* Make sure you stick to the coding style that is used in surrounding code
+  * you can use `pylint --msg-template="{path}:{line}: [{msg_id}({symbol}),
+    {obj}] {msg}" tlslite > pylint_report.txt; diff-quality --violations=pylint
+    pylint_report.txt` to see if your changes (and only your changes) do not
+    violate the general guidelines (alternatively you can just run
+    `make test` as described
+    below).
+* Make commits of logical units and describe them properly in commits
+  * When creating a comment, keep the first line short and separate it from
+    the rest by whiteline
+  * See also [OpenStack guide](https://wiki.openstack.org/wiki/GitCommitMessages)
+    for general good ideas about git commit messages
+* Check for unnecessary whitespace with `git diff --check` before committing
+* Generally newly submitted code should have test coverage so that it can
+  be clearly shown that it works correctly.
+  * inspect results of running `coverage` reported by `make test`
+  * pull requests with code refactoring of code that does not have test
+    coverage should have test coverage of the code added first (as separate
+    commit or separate PR)
+* Assure nothing is broken by running all tests using `make test`
```

Comment:
Again nothing is broken here is quite ambiguous. May worth a separate section on what is tested, and what someone who would like to extend the testsuite should do.